utils.go

```diff
@@ -172,10 +172,7 @@ func getCloudMetadata() *model.Cloud {
 	// package initialisation time. Instead, we defer until it is
 	// first requested by the tracer.
 	cloudMetadataOnce.Do(func() {
-		var logger apmcloudutil.Logger
-		if apmlog.DefaultLogger != nil {
-			logger = apmlog.DefaultLogger
-		}
+		logger := apmlog.DefaultLogger()
```

Comment:
true. looking at the code, this is the only time that any value is assigned to `logger`, so all subsequent code has nil checks, making this functionally equivalent.

It would end up being more code, but I'd be fine adding some sort of `NoopLogger` and having that be the initial type, and then re-introducing the nil check on `apmlog.DefaultLogger()`? Or just undoing the change :shrug:

Reply:
Actually, I'm just going to revert this. It's not really part of the goal of the PR (enabling `-race` on tests)